### PR TITLE
CU-8684xqn30 - Allow using Pydantic 2.0

### PR DESCRIPTION
**Pull request by roguh**
The new version of Pydantic moved the `pydantic.BaseSettings` class into the new `pydantic_settings` package. In this PR, I simply import `BaseSettings` from `pydantic_settings`, and fallback to `pydantic.BaseSettings` if there's an error importing `pydantic_settings`. This package can now be used by clients that depend on Pydantic 1.X or 2.X+.

### CODE REVIEW

**Comment by roguh** [comment]
I decided to use Pydantic 2.0 in the new site-id-cache microservice for this card: https://app.clickup.com/t/8684xqn30

**Comment by kungfuchicken** [comment]
Task linked: [CU-8684xqn30 [E] Cloud WebSocket Proxy (OCPP Router) can route EVSE connection requests](https://app.clickup.com/t/8684xqn30)

File: src/powerflex_logging_utilities/json_formatter.py [hunk 1]
```diff
@@ -13,8 +13,8 @@ class JsonFormatter(jsonlogger.JsonFormatter):  # type: ignore
     - Replaces non-string keys in the log record with their __str__ representation
     """
 
-    def __init__(self, fmt: Optional[str] = DEFAULT_LOG_FORMAT, **kwargs: Any):
-        super().__init__(fmt=fmt, **kwargs)
+    def __init__(self, fmt: Optional[str] = DEFAULT_LOG_FORMAT, **kwargs: Any) -> None:
```
Comment (nicoLedesma) [inline src/powerflex_logging_utilities/json_formatter.py:16]: Is it really necessary to return None?

**Review by nicoLedesma — APPROVED** [review]
LGTM. 👍

File: src/powerflex_logging_utilities/json_formatter.py (hunk 1 above)
Comment (roguh) [reply src/powerflex_logging_utilities/json_formatter.py:16]: This is required for mypy's strict type checking.

**Pull request merged**